reader-osm/src/test/java/com/graphhopper/GraphHopperIT.java

```diff
@@ -1455,50 +1455,50 @@ public void simplifyWithInstructionsAndPathDetails() {
+        assertInstruction(instructions.get(1), "B 85", "[4, 18[", 14, 14);
+        // via instructions have length = 0, but the point list must not be empty!
+        assertInstruction(instructions.get(2), "", "[18, 19[", 0, 1);
+        assertInstruction(instructions.get(3), "B 85", "[18, 35[", 17, 17);
+        assertInstruction(instructions.get(4), "", "[35, 37[", 2, 2);
+        assertInstruction(instructions.get(5), "KU 18", "[37, 40[", 3, 3);
+        assertInstruction(instructions.get(6), "St 2189", "[40, 41[", 1, 1);
+        assertInstruction(instructions.get(7), "", "[41, 43[", 2, 2);
+        // finish instructions have length = 0, but the point list must not be empty!
+        assertInstruction(instructions.get(8), "", "[43, 44[", 0, 1);
 
         // check max speeds
         List<PathDetail> speeds = path.getPathDetails().get("max_speed");
         assertDetail(speeds.get(0), "null [0, 4]");
         assertDetail(speeds.get(1), "70.0 [4, 7]");
-        assertDetail(speeds.get(2), "100.0 [7, 37]");
-        assertDetail(speeds.get(3), "80.0 [37, 38]");
-        assertDetail(speeds.get(4), "null [38, 43]");
-        assertDetail(speeds.get(5), "50.0 [43, 44]");
-        assertDetail(speeds.get(6), "null [44, 46]");
+        assertDetail(speeds.get(2), "100.0 [7, 34]");
+        assertDetail(speeds.get(3), "80.0 [34, 35]");
+        assertDetail(speeds.get(4), "null [35, 40]");
+        assertDetail(speeds.get(5), "50.0 [40, 41]");
+        assertDetail(speeds.get(6), "null [41, 43]");
 
         // check street_names
         List<PathDetail> streetNames = path.getPathDetails().get("street_name");
         assertDetail(streetNames.get(0), "KU 11 [0, 4]");
-        assertDetail(streetNames.get(1), "B 85 [4, 38]");
-        assertDetail(streetNames.get(2), " [38, 40]");
-        assertDetail(streetNames.get(3), "KU 18 [40, 43]");
-        assertDetail(streetNames.get(4), "St 2189 [43, 44]");
-        assertDetail(streetNames.get(5), " [44, 46]");
+        assertDetail(streetNames.get(1), "B 85 [4, 35]");
+        assertDetail(streetNames.get(2), " [35, 37]");
+        assertDetail(streetNames.get(3), "KU 18 [37, 40]");
+        assertDetail(streetNames.get(4), "St 2189 [40, 41]");
+        assertDetail(streetNames.get(5), " [41, 43]");
     }
 
     private void assertInstruction(Instruction instruction, String expectedName, String expectedInterval, int expectedLength, int expectedPoints) {
         assertEquals(expectedName, instruction.getName());
-        if (instruction.getPoints() instanceof ShallowImmutablePointList) {
-            assertEquals(expectedInterval, ((ShallowImmutablePointList) instruction.getPoints()).getIntervalString());
-        }
+        assertEquals(expectedInterval, ((ShallowImmutablePointList) instruction.getPoints()).getIntervalString());
         assertEquals(expectedLength, instruction.getLength());
         assertEquals(expectedPoints, instruction.getPoints().size());
```

Comment:
Compared to #1782 here I am now testing the instruction length and instructions' point list size separately, because they are different for via and finish instructions.